pandas/core/algorithms.py

```diff
@@ -547,10 +566,12 @@ def mode(values):
         constructor = Series
 
     dtype = values.dtype
-    if is_integer_dtype(values):
+    if is_signed_integer_dtype(values):
         values = _ensure_int64(values)
         result = constructor(sorted(htable.mode_int64(values)), dtype=dtype)
```

Comment:
and it can be at the end of the function (and actually you can move the boiler plate to the end of the function)
(e.g. the construct / sort)